### PR TITLE
feat: websocket server add open telemetry

### PR DESCRIPTION
**Pull request by ntheile**

### CODE REVIEW

File: src/servers/ws-server.ts [hunk 1]
```diff
@@ -42,151 +55,189 @@ const getContext = async (
...
+
+            const operationAST = getOperationAST(args.document, args.operationName)
+            if (!operationAST) {
+              // returning `GraphQLError[]` sends an `ErrorMessage` and stops the subscription
+              recordExceptionInCurrentSpan({
+                error: new GraphQLError("Unable to identify operation"),
+                level: ErrorLevel.Warn,
+              })
+              return [new GraphQLError("Unable to identify operation")]
+            }
```
Comment (ntheile) [inline src/servers/ws-server.ts:146]: @nicolasburtey ,

What do you recommend as the default error level for these?

Comment (nicolasburtey) [reply src/servers/ws-server.ts:146]: `warn` sounds good

File: src/servers/ws-server.ts [hunk 2]
```diff
@@ -42,151 +55,189 @@ const getContext = async (
...
+
+            // handle mutation and query requests
```
Comment (nicolasburtey) [inline src/servers/ws-server.ts:148]:
```suggestion
            // decline mutation and query requests
```

File: src/servers/ws-server.ts [hunk 3]
```diff
@@ -42,151 +55,189 @@ const getContext = async (
...
+              addAttributesToCurrentSpan({ "ws.authType": "bearer" })
+              return authorizedContexts[ctx.connectionParams?.Authorization]
+            }
+            // anon context
+            addAttributesToCurrentSpan({ "ws.authType": "anon" })
+            const context = await getContext(ctx)
+            return context
+          },
+          onNext: (ctx, msg, args, result) => {
+            baseLogger.debug("Next", { ctx, msg, args, result })
```
Comment (nicolasburtey) [inline src/servers/ws-server.ts:224]: this is probably where we should have some span events been register to honeycomb (in addition or replacement of the logger) if we want to have more meaning opentelemetry insights.

**Review by nicolasburtey — COMMENTED** [review]
few comments, looks good otherwise

**Review by nicolasburtey — APPROVED** [review]

**Pull request merged**